### PR TITLE
Always serve signed files (bug 1107842)

### PR DESCRIPTION
**Pull request by olivier-m**

### CODE REVIEW

**Comment by clouserw** [comment]
@magopian - can you take over this patch?

File: apps/versions/tests.py [hunk 1]
```diff
@@ -710,6 +716,60 @@ def test_beta_addon_beta_file(self):
...
+
+        fp = zipfile.ZipFile(self.file.file_path, 'w')
+        fp.writestr('install.rdf', '<?xml version="1.0"?><RDF></RDF>')
+        fp.close()
+
+    def tearDown(self):
+        super(TestDownloadsSigned, self).tearDown()
+        os.unlink(self.file.signed_file_path)
+
+    @mock.patch.object(settings, 'SIGNING_SERVER_ACTIVE', True)
```
Comment (magopian) [inline apps/versions/tests.py]: why not use [override_settings](https://docs.djangoproject.com/en/1.7/topics/testing/tools/#django.test.override_settings)?

File: apps/versions/tests.py [hunk 2]
```diff
@@ -710,6 +716,60 @@ def test_beta_addon_beta_file(self):
...
+        eq_(self.addon.status, amo.STATUS_PUBLIC)
+        eq_(self.file.status, amo.STATUS_PUBLIC)
+
+        with mock.patch('lib.crypto.packaged.sign') as mock_sign:
+            self.client.get(self.file_url)
+            assert mock_sign.called
+
+        self.assert_served_signed(self.client.get(self.file_url))
+
+        with mock.patch('lib.crypto.packaged.sign') as mock_sign:
```
Comment (magopian) [inline apps/versions/tests.py]: Should those two lines be removed?

File: apps/versions/tests.py [hunk 3]
```diff
@@ -710,6 +716,60 @@ def test_beta_addon_beta_file(self):
...
+        self.file.update(status=amo.STATUS_NOMINATED)
+        self.assert_served_signed(self.client.get(self.file_url))
+
+    @mock.patch.object(settings, 'SIGNING_SERVER_ACTIVE', True)
+    @mock.patch.object(packaged, 'sign', mock_sign)
+    def test_public_signed_once(self):
+        eq_(self.addon.status, amo.STATUS_PUBLIC)
+        eq_(self.file.status, amo.STATUS_PUBLIC)
+
+        with mock.patch('lib.crypto.packaged.sign') as mock_sign:
```
Comment (magopian) [inline apps/versions/tests.py]: mocking a mock? This line isn't needed, is it?

File: apps/versions/views.py [hunk 4]
```diff
@@ -94,6 +96,17 @@ def download_file(request, file_id, type=None):
     attachment = (type == 'attachment' or not request.APP.browser)
 
     loc = file.get_mirror(addon, attachment=attachment)
+    signed_loc = file.get_mirror(addon, attachment=attachment,
+                                 media_root='signed_addons')
+
+    if storage.exists(file.signed_file_path):
+        loc = signed_loc
+    elif settings.SIGNING_SERVER_ACTIVE and file.can_be_signed():
+            addon.sign_version_files(file.version.pk)
```
Comment (magopian) [inline apps/versions/views.py]: This means that lines 104 to 108 should be removed.

**Comment by magopian** [comment]
Taking over on this PR, opened #424

**Pull request closed without merging**